lib/trino-filesystem/src/main/java/io/trino/filesystem/fileio/BulkOperationsFileIO.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.fileio;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.filesystem.TrinoFileSystem;
+import org.apache.iceberg.io.BulkDeletionFailureException;
+import org.apache.iceberg.io.SupportsBulkOperations;
+
+import java.io.IOException;
+
+public class BulkOperationsFileIO
+        extends ForwardingFileIo
+        implements SupportsBulkOperations
+{
+    public BulkOperationsFileIO(TrinoFileSystem fileSystem)
+    {
+        super(fileSystem);
+    }
+
+    @Override
+    public void deleteFiles(Iterable<String> pathsToDelete)
+            throws BulkDeletionFailureException
+    {
+        ImmutableList.Builder<String> pathList = ImmutableList.builder();
+        pathsToDelete.forEach(path -> pathList.add(path));
+        try {
+            fileSystem.deleteFiles(pathList.build());
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Throwing `RuntimeException` is different from the interface's expectation. It would be nice to throw `BulkDeletionFailureException` or leave a comment.

Reply:
`BulkDeletionFailureException` requires the number of files that failed to delete. Currently `fileSystem.deleteFiles` does not give such information. I will prepare a PR for the same.